docs/design-principles.md

```diff
@@ -0,0 +1,94 @@
+
+### Arch principles
+
+#### `r_rolename_` prefixed variables
+
+**Aim**: All variables used within a role are `r_rolename` and/or `l_` prefixed
+
+**Justification**:
+* To be clear about scope and origin of each variable used within a role:
+  * all local variables of a role prefixed with `l_` string
+  * all role variables seen outside of a role prefixed with `r_rolename_` string
+* To define entry points of a role with an accepted list of role variables, each one prefixed with `r_rolename_` string
+* Eliminate all global variables and make a role self-contained
+
+Example:
+
+**Before:**
+
+```yaml
+# openshift_node/tasks/main.yml
+- name: pull node image
+  command: >
+    docker pull {{ openshift_node_image_name }}:{{ openshift_node_image_tag }}
+  register: docker_pull_result
+```
+
+**After:**
+
+```yaml
+# openshift_node/defaults/main.yml
+r_openshift_node_image_name: openshift-node
+r_openshift_node_image_tag: v3.6.2
+
+# openshift_node/tasks/main.yml
+- name: pull node image
+  command: >
+    docker pull {{ r_openshift_node_image_name }}:{{ r_openshift_node_image_tag }}
+  register: l_docker_pull_result
+```
+
+**Role inlucsion**
+
+**Aim**: Use `include_role` instead of `roles`
+
+**Reason**: Eliminate concept of `pre-tasks`, `roles`, `tasks` and `post-tasks`
+
+Example:
+
+**Before:**
+
+```yaml
+- name:
```

Comment:
Since this is an example of how we want things to be done I think it would be prudent to include a value for the `name:` of this play